File: frontend/src/lib/chains.ts
import { mainnet, polygon, bsc, arbitrum, base } from "wagmi/chains";

export const SUPPORTED_CHAINS = [mainnet, polygon, bsc, arbitrum, base] as const;

export const CHAIN_ID_MAP: Record<number, string> = {
  1: "ethereum",
  137: "polygon",
  56: "bsc",
  42161: "arbitrum",
  8453: "base",
};

export const CHAIN_NAME_MAP: Record<string, string> = {
  ethereum: "Ethereum",
  polygon: "Polygon",
  bsc: "BNB Chain",
  arbitrum: "Arbitrum",
  base: "Base",
};

export function chainIdToSlug(chainId: number): string {
  return CHAIN_ID_MAP[chainId] || "ethereum";
}

File: backend/app/utils/chains.py
from dataclasses import dataclass

@dataclass(frozen=True)
class ChainInfo:
    id: str
    name: str
    alchemy_network: str
    coingecko_platform: str
    chain_id: int


CHAINS: dict[str, ChainInfo] = {
    "ethereum": ChainInfo(
        id="ethereum",
        name="Ethereum",
        alchemy_network="eth-mainnet",
        coingecko_platform="ethereum",
        chain_id=1,
    ),
    "polygon": ChainInfo(
        id="polygon",
        name="Polygon",
        alchemy_network="polygon-mainnet",
        coingecko_platform="polygon-pos",
        chain_id=137,
    ),
    "bsc": ChainInfo(
        id="bsc",
        name="BNB Smart Chain",
        alchemy_network="bnb-mainnet",
        coingecko_platform="binance-smart-chain",
        chain_id=56,
    ),
    "arbitrum": ChainInfo(
        id="arbitrum",
        name="Arbitrum One",
        alchemy_network="arb-mainnet",
        coingecko_platform="arbitrum-one",
        chain_id=42161,
    ),
    "base": ChainInfo(
        id="base",
        name="Base",
        alchemy_network="base-mainnet",
        coingecko_platform="base",
        chain_id=8453,
    ),
}

SUPPORTED_CHAIN_IDS = list(CHAINS.keys())


def get_chain(chain_id: str) -> ChainInfo:
    chain = CHAINS.get(chain_id)
    if not chain:
        raise ValueError(f"Unsupported chain: {chain_id}. Supported: {SUPPORTED_CHAIN_IDS}")
    return chain
